File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/AnswerResults.java
package tech.zolhungaj.amqapi.servercommands.gameroom.game;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandTypeOld;
import tech.zolhungaj.amqapi.servercommands.objects.PlayerAnswerResult;
import tech.zolhungaj.amqapi.servercommands.objects.ProgressBarState;
import tech.zolhungaj.amqapi.servercommands.objects.SongInfo;

import java.util.List;
import java.util.Map;

public record AnswerResults(
        @Json(name = "watched")
        boolean isWatched,
        @Json(name = "groupMap")
        Map<String, List<Integer>> groupMap,
        @Json(name = "players")
        List<PlayerAnswerResult> players,
        @Json(name = "songInfo")
        SongInfo songInfo,
        @Json(name = "progressBarState")
        ProgressBarState progressBarState
) implements Command {

        @Override
        public String commandName() {
            return CommandTypeOld.ANSWER_RESULTS.commandName;
        }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/AnswerResults.kt
package tech.zolhungaj.amqapi.servercommands.gameroom.game

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.PlayerAnswerResult
import tech.zolhungaj.amqapi.servercommands.objects.ProgressBarState
import tech.zolhungaj.amqapi.servercommands.objects.SongInfo

@JvmRecord
@CommandType("answer results")
data class AnswerResults(
    @Json(name = "watched")
    val isWatched: Boolean,
    @Json(name = "groupMap")
    val groupMap: Map<String, List<Int>>,
    @Json(name = "players")
    val players: List<PlayerAnswerResult>,
    @Json(name = "songInfo")
    val songInfo: SongInfo,
    @Json(name = "progressBarState")
    val progressBarState: ProgressBarState
)
